File: scala-play-server/app/api/DefaultApiImpl.scala
package api

import model.RequestRoutes
import model.RequestSupportedLocations
import model.RequestTimeFilter
import model.RequestTimeFilterFast
import model.RequestTimeFilterPostcodeDistricts
import model.RequestTimeFilterPostcodeSectors
import model.RequestTimeFilterPostcodes
import model.RequestTimeMap
import model.ResponseError
import model.ResponseGeocoding
import model.ResponseMapInfo
import model.ResponseRoutes
import model.ResponseSupportedLocations
import model.ResponseTimeFilter
import model.ResponseTimeFilterFast
import model.ResponseTimeFilterPostcodeDistricts
import model.ResponseTimeFilterPostcodeSectors
import model.ResponseTimeFilterPostcodes
import model.ResponseTimeMap

/**
  * Provides a default implementation for [[DefaultApi]].
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2020-03-18T07:55:16.100Z[Etc/UTC]")
class DefaultApiImpl extends DefaultApi {
  /**
    * @inheritdoc
    */
  override def geocodingReverseSearch(lat: Double, lng: Double, withinPeriodcountry: Option[String]): ResponseGeocoding = {
    // TODO: Implement better logic

    ResponseGeocoding("", List.empty[ResponseGeocodingGeoJsonFeature])
  }

  /**
    * @inheritdoc
    */
  override def geocodingSearch(query: String, focusPeriodlat: Option[Double], focusPeriodlng: Option[Double], withinPeriodcountry: Option[String]): ResponseGeocoding = {
    // TODO: Implement better logic

    ResponseGeocoding("", List.empty[ResponseGeocodingGeoJsonFeature])
  }

  /**
    * @inheritdoc
    */
  override def mapInfo(): ResponseMapInfo = {
    // TODO: Implement better logic

    ResponseMapInfo(List.empty[ResponseMapInfoMap])
  }

  /**
    * @inheritdoc
    */
  override def routes(requestRoutes: RequestRoutes): ResponseRoutes = {
    // TODO: Implement better logic

    ResponseRoutes(List.empty[ResponseRoutesResult])
  }

  /**
    * @inheritdoc
    */
  override def supportedLocations(requestSupportedLocations: RequestSupportedLocations): ResponseSupportedLocations = {
    // TODO: Implement better logic

    ResponseSupportedLocations(List.empty[ResponseSupportedLocation], List.empty[String])
  }

  /**
    * @inheritdoc
    */
  override def timeFilter(requestTimeFilter: RequestTimeFilter): ResponseTimeFilter = {
    // TODO: Implement better logic

    ResponseTimeFilter(List.empty[ResponseTimeFilterResult])
  }

  /**
    * @inheritdoc
    */
  override def timeFilterFast(requestTimeFilterFast: RequestTimeFilterFast): ResponseTimeFilterFast = {
    // TODO: Implement better logic

    ResponseTimeFilterFast(List.empty[ResponseTimeFilterFastResult])
  }

  /**
    * @inheritdoc
    */
  override def timeFilterPostcodeDistricts(requestTimeFilterPostcodeDistricts: RequestTimeFilterPostcodeDistricts): ResponseTimeFilterPostcodeDistricts = {
    // TODO: Implement better logic

    ResponseTimeFilterPostcodeDistricts(List.empty[ResponseTimeFilterPostcodeDistrictsResult])
  }

  /**
    * @inheritdoc
    */
  override def timeFilterPostcodeSectors(requestTimeFilterPostcodeSectors: RequestTimeFilterPostcodeSectors): ResponseTimeFilterPostcodeSectors = {
    // TODO: Implement better logic

    ResponseTimeFilterPostcodeSectors(List.empty[ResponseTimeFilterPostcodeSectorsResult])
  }

  /**
    * @inheritdoc
    */
  override def timeFilterPostcodes(requestTimeFilterPostcodes: RequestTimeFilterPostcodes): ResponseTimeFilterPostcodes = {
    // TODO: Implement better logic

    ResponseTimeFilterPostcodes(List.empty[ResponseTimeFilterPostcodesResult])
  }

  /**
    * @inheritdoc
    */
  override def timeMap(requestTimeMap: RequestTimeMap): ResponseTimeMap = {
    // TODO: Implement better logic

    ResponseTimeMap(List.empty[ResponseTimeMapResult])
  }
}

File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/api/DefaultApiImpl.java
package org.openapitools.vertxweb.server.api;

import org.openapitools.vertxweb.server.model.RequestRoutes;
import org.openapitools.vertxweb.server.model.RequestSupportedLocations;
import org.openapitools.vertxweb.server.model.RequestTimeFilter;
import org.openapitools.vertxweb.server.model.RequestTimeFilterFast;
import org.openapitools.vertxweb.server.model.RequestTimeFilterPostcodeDistricts;
import org.openapitools.vertxweb.server.model.RequestTimeFilterPostcodeSectors;
import org.openapitools.vertxweb.server.model.RequestTimeFilterPostcodes;
import org.openapitools.vertxweb.server.model.RequestTimeMap;
import org.openapitools.vertxweb.server.model.ResponseError;
import org.openapitools.vertxweb.server.model.ResponseGeocoding;
import org.openapitools.vertxweb.server.model.ResponseMapInfo;
import org.openapitools.vertxweb.server.model.ResponseRoutes;
import org.openapitools.vertxweb.server.model.ResponseSupportedLocations;
import org.openapitools.vertxweb.server.model.ResponseTimeFilter;
import org.openapitools.vertxweb.server.model.ResponseTimeFilterFast;
import org.openapitools.vertxweb.server.model.ResponseTimeFilterPostcodeDistricts;
import org.openapitools.vertxweb.server.model.ResponseTimeFilterPostcodeSectors;
import org.openapitools.vertxweb.server.model.ResponseTimeFilterPostcodes;
import org.openapitools.vertxweb.server.model.ResponseTimeMap;

import org.openapitools.vertxweb.server.ApiResponse;

import io.vertx.core.Future;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.handler.impl.HttpStatusException;

import java.util.List;
import java.util.Map;

// Implement this class

public class DefaultApiImpl implements DefaultApi {
    public Future<ApiResponse<ResponseGeocoding>> geocodingReverseSearch(Double lat, Double lng, String withinCountry) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseGeocoding>> geocodingSearch(String query, Double focusLat, Double focusLng, String withinCountry) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseMapInfo>> mapInfo() {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseRoutes>> routes(RequestRoutes requestRoutes) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseSupportedLocations>> supportedLocations(RequestSupportedLocations requestSupportedLocations) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeFilter>> timeFilter(RequestTimeFilter requestTimeFilter) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeFilterFast>> timeFilterFast(RequestTimeFilterFast requestTimeFilterFast) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeFilterPostcodeDistricts>> timeFilterPostcodeDistricts(RequestTimeFilterPostcodeDistricts requestTimeFilterPostcodeDistricts) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeFilterPostcodeSectors>> timeFilterPostcodeSectors(RequestTimeFilterPostcodeSectors requestTimeFilterPostcodeSectors) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeFilterPostcodes>> timeFilterPostcodes(RequestTimeFilterPostcodes requestTimeFilterPostcodes) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<ResponseTimeMap>> timeMap(RequestTimeMap requestTimeMap) {
        return Future.failedFuture(new HttpStatusException(501));
    }

}
